tests/test_middleware_interface.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of prompt_prototype.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os.path
+import tempfile
+import unittest
+
+from activator.middleware_interface import MiddlewareInterface
+from activator.visit import Visit
+
+
+class MiddlewareInterfaceTest(unittest.TestCase):
+    def setUp(self):
+        input_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), "data")
+        instrument = "TestCam"
+        self.output_dir = tempfile.TemporaryDirectory()
+        self.interface = MiddlewareInterface(input_dir, self.output_dir, instrument)
+        self.next_visit = Visit(instrument,
+                                detector=1,
+                                group=1,
+                                snaps=1,
+                                filter="r",
+                                ra=10,
+                                dec=20,
+                                kind="BIAS")
+        self.log_prefix = "INFO:activator.middleware_interface:"
+
+    def test_init(self):
+        """Basic tests of initializing an interface object.
+        """
+        # did we get a useable butler instance?
+        self.assertIn("/tmp/butler-", self.interface.dest.datastore.root.path)
+        # Ideas for things to test:
+        # * On init, does the right kind of butler get created, with the right
+        #   collections, etc?
+        # * On init, is the local butler repo purely in memory?
+
+    def test_prep_butler(self):
+        with self.assertLogs("activator.middleware_interface", level="INFO") as cm:
+            self.interface.prep_butler(self.next_visit)
+        msg = f"{self.log_prefix}Preparing Butler for visit '{self.next_visit}'"
+        self.assertEqual(cm.output, [msg])
```

Comment:
`assertIn`? I think my changes on [DM-33931](https://jira.lsstcorp.org/browse/DM-33931) may make it hard to keep up this test style.

Reply:
I hope to get rid of all of these assertLogs tests when I add butler API calls to do the work, so this is a short stopgap.